openghg/retrieve/ceda/_retrieve.py

```diff
@@ -222,6 +228,8 @@ def local_retrieve_surface(
 
     to_store = {key: {"data": dataset, "metadata": metadata}}
 
-    ObsSurface.store_data(data=to_store)
+    bucket = get_writable_bucket(name=store)
+    with ObsSurface(bucket=bucket) as obs:
```

Comment:
What problem is the context manager API solving do you think (e.g. rather than `ObsSurface.store_data(data=to_store, bucket=bucket)`)? Just wondering as it seems to be adding complexity when calling the `ObsSurface.store_data` method (and others) now this is no longer a static method?

Reply:
It has simplified the code in the base class quite a lot and makes opening and closing the metastore easier, removing the need for `load_metadatastore`, `metadatastore.close` etc across the code. It won't have to be used in many places but I think adds quite a bit of safety and ensures things are written correctly.